### PR TITLE
Create random JWT secret if none provided

### PR DESCRIPTION
**Pull request by bmallred**
Resolves truetandem/e-QIP-prototype#2773

### CODE REVIEW

**Comment by ryanhofdotgov** [comment]
Thanks for the quick turnaround. 

* Was there something specific (e.g., finding from someone else) that lead you to increase the strength of the hash algorithm (HS256 to HS512)? 

* UUID v4 only generates 122-bits of randomness. For HS256, NIST says you need at least 256-bit key size (hopefully all random).  For HS5512, it needs to be 512-bits of randomness ideally. Here is reference:
https://tools.ietf.org/html/rfc7518#section-3.2

Something like this generates 256-bits of randomness in base64 (base64 as convenience, since we are inputing key as string in environment):
`openssl rand -base64 32`
Here is 512-bits:
`openssl rand -base64 64 | tr -d '\n'`

* If UUID call fails, using timestamp as the JWT secret is not "failing secure" since the timestamp can be guessed trivially.

**Comment by bmallred** [comment]
@ryanhofdotgov some benchmarks between 256 and 512 bits.

 - **GOOS**: linux
 - **GOARCH**: amd64
 - **PKG**: github.com/18F/e-QIP/prototype/api/jwt

## With benchmark minimum time at default (1 second)

### Command

```shell
go test -bench=.
```

### Results 

| Benchmark | Executions | Avg. Time
| --- | ----: | ----:
| BenchmarkRandomSecret256-8 | 300000 | 4277 ns/op
| BenchmarkRandomSecret512-8 | 300000 | 4773 ns/op

## With benchmark minimum time at 20 seconds

### Command

```shell
go test -bench=. -benchtime=20s
```

### Results 

| Benchmark | Executions | Avg. Time
| --- | ----: | ----:
| BenchmarkRandomSecret256-8 | 10000000 | 4419 ns/op
| BenchmarkRandomSecret512-8 | 5000000 | 5001 ns/op

**Comment by bmallred** [comment]
@floresj I think this one is ready for review as well :smile:

**Comment by floresj** [comment]
@bmallred schweet

File: api/jwt/token_test.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	"time"
+
+	jwt "github.com/dgrijalva/jwt-go"
+)
+
+func TestKnownSecret(t *testing.T) {
+	envar := "JWT_SECRET"
+	old := os.Getenv(envar)
+
+	expected := "too many secrets"
```
Comment (floresj) [inline api/jwt/token_test.go]: But I like secrets.

Comment (bmallred) [reply api/jwt/token_test.go]: Quote from Sneakers: https://www.youtube.com/watch?v=GutJf9umD9c

Comment (floresj) [reply api/jwt/token_test.go]: TIL Sneakers 🎉

**Pull request merged**